### PR TITLE
[GH-15680] Adding installation disclaimer in documentation

### PR DESCRIPTION
**Pull request by hannah-tillman**
For #15680 

This PR adds a note that by default installation is open and to follow our security guidelines if you want a secure installation.

### CODE REVIEW

File: h2o-dist/index.html [hunk 1]
```diff
@@ -562,6 +562,7 @@ <h1><span>H<sub>2</sub>O</span></h1>
             </div>
             <h2>Get started with H<sub>2</sub>O in 3 easy steps</h2>
             <p>1. Download H<sub>2</sub>O. This is a zip file that contains everything you need to get started.</p>
+            <p> By default, this setup is open. If you want to secure your installation, follow <a href="https://docs.h2o.ai/h2o/latest-stable/h2o-docs/security.html" target="_blank"> Security Guidelines.</a></p>
```
Comment (mmalohlava) [inline h2o-dist/index.html:565]: @hannah-tillman can we put it into some kind of "info" box - let's consult @jefffohl maybe there is already a container ready to do that.

File: h2o-dist/index.html [hunk 2]
```diff
@@ -817,6 +821,8 @@ <h2>User Documentation</h2>
           <div id="user_documentation"></div>
           <h2>Developer Documentation</h2>
           <div id="developer_documentation"></div>
+          <h2>Security Documentation</h2>
+          <p><a href="https://docs.h2o.ai/h2o/latest-stable/h2o-docs/security.html" target="_blank"> Security Guidelines</a></p>
```
Comment (mmalohlava) [inline h2o-dist/index.html:825]: minor - it should point to release specific security guidelines.

**Review by mmalohlava — CHANGES_REQUESTED** [review]
@hannah-tillman Just a change for download page.

Also may i ask you to generate the docs and share the zip on the PR please.

File: h2o-dist/index.html (hunk 2 above)
Comment (hannah-tillman) [reply h2o-dist/index.html:825]: Perhaps I am missing something, but I don't believe there's security entirely specific to just releases here? Seems like different types of initialization, but I could be misunderstanding. Is there a specific section you have in mind?

File: h2o-dist/index.html (hunk 1 above)
Comment (jefffohl) [reply h2o-dist/index.html:565]: There doesn't appear to be one - I can add one. Is it OK for me to push changes to this branch @hannah-tillman ? Or would you prefer a PR targeting this branch?

Comment (hannah-tillman) [reply h2o-dist/index.html:565]: @jefffohl Please go ahead and make changes directly here. Thank you so much!

File: h2o-dist/index.html [hunk 3]
```diff
@@ -247,6 +247,35 @@
       .pre {
           white-space: pre;
       }
+      
+      .callout {
```
Comment (jefffohl) [inline h2o-dist/index.html:251]: I added callout styles. Here is what they look like:

Screenshot 2023-08-10 at 2 08 27 PM

Comment (hannah-tillman) [reply h2o-dist/index.html:251]: @jefffohl Thank you -- this looks so nice! Is it cool if I update the color so it's less panicky? Perhaps, these blues? 
Screen Shot 2023-08-10 at 4 30 27 PM

Comment (jefffohl) [reply h2o-dist/index.html:251]: Yes, sure - no problem - looks nice.

File: h2o-dist/index.html [hunk 4]
```diff
@@ -258,23 +258,23 @@
       
       .callout--warning {
```
Comment (jefffohl) [inline h2o-dist/index.html:259]: nit pick: With the blue color, I recommend changing the class name from `.callout--warning` to `.callout--info`

**Review by jefffohl — COMMENTED** [review]
Just a nit:

File: h2o-dist/index.html (hunk 4 above)
Comment (hannah-tillman) [reply h2o-dist/index.html]: ah, good point. Done 👍

**Review by jefffohl — APPROVED** [review]
LGTM (from a CSS/HTML perspective) - thanks!

**Review by wendycwong — APPROVED** [review]

File: h2o-dist/index.html (hunk 2 above)
Comment (mn-mikke) [reply h2o-dist/index.html:854]: There might be a minor changes in h2o documentation for older h2o versions. So if you share the download page for an older h2o. (for this and subsequent versions in future). The documentation doesn't have to be precisely accurate.

**Review by mn-mikke — APPROVED** [review]
LGTM

**Pull request merged**